src/lib/models/transaction.js

```diff
@@ -19,6 +19,10 @@ const statuses = {
   error: 'error'
 };
 
+const getDescriptionFromChargeElement = chargeElement => {
+  return chargeElement.description || chargeElement.purposeUse.name;
+};
+
 const getTwoPartTariffTransactionDescription = (transaction) => {
```

Comment:
I wonder if this function would need to change slightly to handle a charge element with no description - presumably it should either say:
  return `${prefix} part ${purpose} charge at ${description}`;
or 
  return `${prefix} part ${purpose} charge`;   // when description empty